rpm/__init__.py

```diff
@@ -79,21 +96,112 @@ def try_path(path: str) -> bool:
     return False
 
 
-def initialize() -> None:
+def reload_module(path: Path, suffixes: List[str]) -> None:
     """
-    Initializes the shim. Tries to load system RPM module and replace itself with it.
+    Reloads the `rpm` module. In case some of the required binary submodules fail to import,
+    tries to import them directly by path using valid extension suffixes.
+
+    Args:
+        path: Absolute path to the `rpm` module.
+        suffixes: List of extension suffixes valid for the Python interpreter associated
+                  with the path.
     """
-    for path in get_system_sitepackages():
-        logger.debug(f"Trying {path}")
+    attempted_modules = []
+    while True:
         try:
-            if try_path(path):
-                logger.debug("Import successfull")
-                return
-        except ShimAlreadyInitializingError:
-            continue
-        except Exception as e:
-            logger.debug(f"Exception: {type(e)}: {e}")
+            importlib.reload(sys.modules[__name__])
+        except ModuleNotFoundError as e:
+            if e.name in attempted_modules:
```

Comment:
Ok, the mypy error means that `e.name` in this exception block can be `None`, so there needs to be a check for that.

Reply:
Yeah I originally had an e.name value check and forgot why. It was for this. Fixed again